manifests/machineconfig.crd.yaml

```diff
@@ -39,3 +39,208 @@ spec:
+                          group:
+                            description: "group object specifies group of the owner"
+                            type: object
+                            properties:
+                              id:
+                                description: "id specifies group ID of the owner"
+                                type: integer
+                              name:
+                                description: "name is the group name of the owner"
+                                type: string
+                          overwrite:
+                            description: "overwrite specifies whether to delete preexisting nodes at the path"
+                            type: boolean
+                systemd:
+                  description: "systemd describes the desired state of the systemd units"
+                  type: object
+                  properties:
+                    units:
+                      description: "units is a list of units to be configured"
+                      type: array
+                      items:
+                        description: "items describes unit configuration"
+                        type: object
+                        properties:
+                          name:
+                            description: "name is the name of the unit. This must be suffixed with a valid unit type (e.g. 'thing.service')"
+                            type: string
+                          enabled:
+                            description: "enabled describes whether or not the service shall be enabled. When true, the service is enabled. When false, the service is disabled. When omitted, the service is unmodified. In order for this to have any effect, the unit must have an install section"
+                            type: boolean
+                          mask:
+                            description: "mask describes whether or not the service shall be masked. When true, the service is masked by symlinking it to /dev/null"
+                            type: boolean
+                          contents:
+                            description: "contents is the contents of the unit"
+                            type: string
+                          dropins:
+                            description: "dropins is the list of drop-ins for the unit"
+                            type: array
+                            items:
+                              description: "items describes unit dropin"
+                              type: object
+                              properties:
+                                contents:
+                                  description: "contents is the contents of the drop-in"
+                                  type: string
+                                name:
+                                  description: "name is the name of the drop-in. This must be suffixed with '.conf'."
+                                  type: string
```

Comment:
do we need:
```
required:
      - spec
```
to make it error out on `oc create`? (following https://github.com/openshift/cluster-version-operator/pull/168/files#diff-9b1e80d552a6692388b5e226ebf4c02cR137)

Reply:
Right, I'll add more `requires` there

Reply:
I think spec is enough - `config`, `kargs`, `fips` can be `anyOf`

Reply:
Lets leave it like this for now - IIUC `anyOf` and detailed type validation is 1.15 feature and not yet enabled in 4.2